lib/charms/layer/kubernetes_master.py

```diff
@@ -141,13 +149,170 @@ def get_cephfs_fsname():
 
 def deprecate_auth_file(auth_file):
     """
-    In 1.19+, file-based authentication was deprecated in favor of cert
+    In 1.19+, file-based authentication was deprecated in favor of webhook
     auth. Write out generic files that inform the user of this.
     """
     csv_file = Path(auth_file)
     csv_file.parent.mkdir(exist_ok=True)
+
+    csv_backup = Path('{}.{}'.format(csv_file, AUTH_BACKUP_EXT))
+    if csv_file.exists() and not csv_backup.exists():
+        csv_file.rename(csv_backup)
```

Comment:
A brand new bootstrap ends up with a `known_tokens.csv.pre-migration` file with an incorrect admin user token in it.  Not a big deal, but could be a bit confusing, so it might be nice to avoid it if we can.

Reply:
Yeah, this happens because sometimes [password_changed](https://github.com/charmed-kubernetes/charm-kubernetes-master/blob/master/reactive/kubernetes_master.py#L525) is called before `setup_leader_auth`.  I haven't been able to come up with a way around this without possibly losing the `config.changed.password` when config is changed post-deployment.

Reply:
I was able to work around this by reconfiguring auth and *not* writing a token in `password_changed`.  We still get the password changed, we just write it as a secret instead of to a file during new deployments.